vyper/semantics/types/base.py

```diff
@@ -25,7 +25,7 @@ def __init__(self, type_):
         self.type_ = type_
 
     def compare_type(self, other):
-        return isinstance(other, self.type_)
+        return isinstance(other, self.type_) or self == other
```

Comment:
This allows for comparison when both self and other are generic types e.g. when it is a bytestring.

Reply:
kind of interesting .. we will probably want to disallow that case, but it's a language change.

Reply:
what is the behavior of `__eq__` on GenericTypeAcceptor?